# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: mozglue/tests/TestBaseProfiler.cpp
# Commit: 3aa39fa769a2
# Full Hash: 3aa39fa769a237babc6d2a09c39a4de58e9e589b
# Author: Gerald Squelart <gsquelart@mozilla.com>
# Date: 2020-10-22 21:51:59
# Description:
#   Bug 1672501 - Initialize scProfilerMainThreadId from profiler_init - r=gregtatum
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D94419
# ==============================================================================

diff -r 766437adb95f -r 3aa39fa769a2 mozglue/tests/TestBaseProfiler.cpp
--- a/mozglue/tests/TestBaseProfiler.cpp	Thu Oct 22 12:46:28 2020 +0000
+++ b/mozglue/tests/TestBaseProfiler.cpp	Thu Oct 22 12:55:13 2020 +0000
@@ -71,31 +71,6 @@
 
 using namespace mozilla;
 
-void TestProfilerUtilities() {
-  printf("TestProfilerUtilities...\n");
-
-  // We'll assume that this test runs in the main thread (which should be true
-  // when called from the `main` function).
-  const int mainThreadId = mozilla::baseprofiler::profiler_current_thread_id();
-
-  MOZ_RELEASE_ASSERT(mozilla::baseprofiler::profiler_main_thread_id() ==
-                     mainThreadId);
-  MOZ_RELEASE_ASSERT(mozilla::baseprofiler::profiler_is_main_thread());
-
-  std::thread testThread([&]() {
-    const int testThreadId =
-        mozilla::baseprofiler::profiler_current_thread_id();
-    MOZ_RELEASE_ASSERT(testThreadId != mainThreadId);
-
-    MOZ_RELEASE_ASSERT(mozilla::baseprofiler::profiler_main_thread_id() !=
-                       testThreadId);
-    MOZ_RELEASE_ASSERT(!mozilla::baseprofiler::profiler_is_main_thread());
-  });
-  testThread.join();
-
-  printf("TestProfilerUtilities done\n");
-}
-
 void TestPowerOfTwoMask() {
   printf("TestPowerOfTwoMask...\n");
 
@@ -3233,7 +3208,6 @@
 }
 
 void TestProfilerDependencies() {
-  TestProfilerUtilities();
   TestPowerOfTwoMask();
   TestPowerOfTwo();
   TestLEB128();
@@ -3308,6 +3282,24 @@
     MOZ_RELEASE_ASSERT(!baseprofiler::profiler_thread_is_being_profiled());
     MOZ_RELEASE_ASSERT(!baseprofiler::profiler_thread_is_sleeping());
 
+    const int mainThreadId =
+        mozilla::baseprofiler::profiler_current_thread_id();
+
+    MOZ_RELEASE_ASSERT(mozilla::baseprofiler::profiler_main_thread_id() ==
+                       mainThreadId);
+    MOZ_RELEASE_ASSERT(mozilla::baseprofiler::profiler_is_main_thread());
+
+    std::thread testThread([&]() {
+      const int testThreadId =
+          mozilla::baseprofiler::profiler_current_thread_id();
+      MOZ_RELEASE_ASSERT(testThreadId != mainThreadId);
+
+      MOZ_RELEASE_ASSERT(mozilla::baseprofiler::profiler_main_thread_id() !=
+                         testThreadId);
+      MOZ_RELEASE_ASSERT(!mozilla::baseprofiler::profiler_is_main_thread());
+    });
+    testThread.join();
+
     printf("profiler_start()...\n");
     Vector<const char*> filters;
     // Profile all registered threads.